src/components/coin-row/TransactionCoinRow.js

```diff
@@ -163,16 +161,15 @@ export default function TransactionCoinRow({ item, ...props }) {
             ? `${headerInfo.type} ${date} ${headerInfo.divider} ${headerInfo.address}`
             : `${headerInfo.type} ${date}`,
         },
-        buttonIndex => {
+        async buttonIndex => {
           const action = buttons[buttonIndex];
           switch (action) {
             case TransactionActions.viewContact:
             case TransactionActions.addToContacts:
               navigate(Routes.MODAL_SCREEN, {
                 address: contactAddress,
-                asset: item,
-                color: contactColor,
-                contact,
+                contactNickname: contact?.nickname,
+                ens: ens,
```

Comment:
i see we're using `contact?.ens` in other places, maybe we can do the same?

Reply:
idk if the concept of storing ens in contacts makes sense to me. when i think of what kind of data should be stored in a contact, this is what comes to mind:

1. data should not be time-sensitive. any amount of time should be able to safely pass between two usages of a particular contact.
2. data should be specific to the user. for example, a user can manually override a contact's nickname, emoji and color. these changes shouldn't affect any other user, so it makes sense why they should be stored locally in a contact.

since the relationship between an eth wallet address and an ens can change at any point in time, i think it's important to fetch this data every time an ens is used to make sure it is still valid. if we do that, then what's the point of storing it locally?

Reply:
if you agree then i think it makes sense to get rid of all instances of ens being stored in a contact

Reply:
hm i agree and we do that for the imported wallets on the app, on startup we get the ens names, maybe another way to go would be to get contacts ens names in a similar way? feels strange to have to fetch the ens name on a useEffect every time you want to use the contact ens

Reply:
about removing the `ens` param from the contact object, is probably a good point, we now only store that value when adding the contact on `onAddOrUpdateContacts` so if the ens changes the contact ens is not changing at all, so is not up to date

Reply:
coming back to `maybe another way to go would be to get contacts ens names in a similar way?`, do we have a use query hook to get this info from the profiles work

Reply:
about your last comment, i was wondering the same thing. if none exists, could be good to make one along with a prefetch fn